frontend/src/components/NewGroup.tsx

```diff
@@ -64,37 +67,61 @@ function NewGroup(props: { onClose: () => void; session: Session | null }) {
   ];
   const [isCreatingGroup, setIsCreatingGroup] = React.useState<boolean>(false); // new state
   const [selectedTab, setSelectedTab] = React.useState<number>(0); // new state
-  const [user, setUser] = useState(null); // new state
+  const [user, setUser] = useState<string | null>(null);
+  const userData = useFetchUserData(props.session);
 
+  const [customAmounts, setCustomAmounts] = useState<Record<string, number>>(
+    {}
+  );
+  const [pricePerMember, setPricePerMember] = useState(Number);
```

Comment:
same as comment with subscriptionCost above, pricePerMember could be a computed property based on the number of users. Don't need it to be inside of state.